source/binding/Bitstream.cpp

```diff
@@ -586,19 +663,20 @@ static bool unpackConcatenation(const StreamingConcatenationExpression& lhs, Pac
                                 size_t& dynamicSize, EvalContext& context,
                                 SmallVector<ConstantValue>* dryRun = nullptr) {
     for (auto stream : lhs.streams()) {
-        if (stream->kind == ExpressionKind::Streaming) {
-            auto& concat = stream->as<StreamingConcatenationExpression>();
+        auto& operand = *stream->operand;
+        if (operand.kind == ExpressionKind::Streaming) {
+            auto& concat = operand.as<StreamingConcatenationExpression>();
             if (dryRun || !concat.sliceSize) {
-                if (!unpackConcatenation(stream->as<StreamingConcatenationExpression>(), iter,
-                                         iterEnd, bitOffset, dynamicSize, context, dryRun)) {
+                if (!unpackConcatenation(operand.as<StreamingConcatenationExpression>(), iter,
```

Comment:
Could just use "concat" here instead of casting again.

Reply:
Yes, I replace/remove three redundant as\<StreamingConcatenationExpression\>() in the file.